terragraph_planner/optimization/topology_operations.py

```diff
@@ -504,125 +502,104 @@ def compute_capex(topology: Topology, params: OptimizerParams) -> Capex:
 
     # Precompute net gain because it does not change during tx power modulation
     # Precomputation can save significant time for large networks
     link_net_gain_map = compute_link_net_gain_map(topology)
 
-    interference_ok_count_old = 0
     if len(active_links) > 0:
-        for _ in range(MAX_LINK_BUDGET_ITERATIONS):
-            # compute interference and SINR
+        for _ in range(max_iterations):
+            # Compute interference and SINR
             analyze_interference(topology, link_net_gain_map)
-            # adjust tx power and link measured for ALL wireless links
+            # Adjust tx power and link measurements
+            tx_power_stable_count = 0
             for link in active_links:
                 tx_data = topology.sites[link.tx_site.site_id].device
                 min_tx_power = tx_data.sector_params.minimum_tx_power
                 rx_data = topology.sites[link.rx_site.site_id].device
                 np_dbm = get_noise_power(rx_data.sector_params)
                 mcs_snr_mbps_map = rx_data.sector_params.mcs_map
 
-                # adjust MCS & capacity based on SINR
-                link.mcs_level, link.capacity = get_link_capacity_from_snr(
+                # Adjust MCS and capacity based on SINR
+                link.mcs_level = get_mcs_from_snr(
                     link.sinr_dbm, mcs_snr_mbps_map
                 )
+                link.capacity = get_link_capacity_from_mcs(
+                    link.mcs_level, mcs_snr_mbps_map
+                )
                 n_and_i_dbm = link.rsl_dbm - link.sinr_dbm
-                # compute the lower Tx power which results the same MCS
+                # Compute the lowest tx power that maintains the same MCS
+                tx_power_prev = link.tx_power
                 link.mcs_level, link.tx_power = adjust_tx_power_with_backoff(
                     mcs_level=link.mcs_level,
                     mcs_snr_mbps_map=mcs_snr_mbps_map,
                     min_tx_power=min_tx_power,
-                    max_tx_power=max_tx_power[link.link_id],
+                    max_tx_power=max_tx_power[link.tx_site.device.device_sku],
                     net_gain_dbi=net_gain[link.link_id],
                     np_dbm=n_and_i_dbm,
                 )
-                # re-compute link budgets based on the new Tx power
+                # Re-compute link measurements from the udpated tx power
                 link.rsl_dbm = get_fspl_based_rsl(
                     link.tx_power, net_gain[link.link_id]
                 )
                 link.snr_dbm = get_snr(link.rsl_dbm, np_dbm)
 
-            # stop iteration if all link interferences converged
-            interference_ok_count = sum(
-                (link.snr_dbm - link.sinr_dbm) < LINK_BUDGET_CONVERGE_THRESHOLD
-                for link in active_links
-            )
+                # Tx power does not change if the MCS with SINR stayed the same
```

Comment:
This changes the convergence criteria to tx power not changing rather than snr close to sinr which might not be possible.